client/src/modules/MentorRegistry/components/MentorRegistryModal.tsx

```diff
@@ -0,0 +1,115 @@
+        return (
+          <Modal
+            style={{ top: 20 }}
+            width={700}
+            open
+            title="Invite Mentor for a Courses"
+            okText="Send Invite"
+            onOk={async e => {
+              e.preventDefault();
+              const values = await form.validateFields().catch(() => null);
+              if (values == null) {
+                return;
+              }
+              handleModalSubmit(values);
+            }}
+            onCancel={() => {
+              onCancel();
+              form.resetFields();
+            }}
+          >
+            <Spin spinning={modalLoading ?? false}>
+              <Form layout="vertical" form={form} initialValues={getInitialValues(modalData.record)}>
+                <Form.Item name="preselectedCourses" label="Pre-Selected Courses" required>
+                  <Select mode="multiple" optionFilterProp="children">
+                    {courses.map(course => (
+                      <Select.Option disabled={course.completed} key={course.id} value={course.id}>
+                        {course.name}
+                      </Select.Option>
+                    ))}
+                  </Select>
+                </Form.Item>
+              </Form>
+            </Spin>
+          </Modal>
+        );
+      case 're-send':
+        return (
+          <Modal
+            width={420}
+            onOk={() => resendConfirmation(modalData.record)}
+            open
+            onCancel={onCancel}
+            okText="Re-send"
+          >
+            <Spin spinning={modalLoading ?? false}>
+              <div style={{ display: 'flex' }}>
+                <ExclamationCircleOutlined style={{ fontSize: '32px', color: '#178df9', marginRight: 16 }} />
+                <Col>
+                  <Typography.Title level={5}>Re-send Invitation for a Courses</Typography.Title>
+                  <Col>Do you want resend invitation for a not accepted courses?</Col>
+                </Col>
+              </div>
+            </Spin>
+          </Modal>
+        );
+      case 'delete':
+        return (
+          <Modal
+            width={420}
+            onOk={() => cancelMentor(modalData.record.githubId)}
+            open
+            onCancel={onCancel}
+            okText="Delete"
+            okButtonProps={{ danger: true, type: 'primary' }}
+          >
+            <Spin spinning={modalLoading ?? false}>
+              <div style={{ display: 'flex' }}>
+                <ExclamationCircleOutlined style={{ fontSize: '32px', color: '#faad14', marginRight: 16 }} />
+                <Col>
+                  <Typography.Title level={5}>Are you sure to delete this Mentor apply?</Typography.Title>
+                  <Col>If you delete mentor's apply you can't restore it.</Col>
+                </Col>
+              </div>
+            </Spin>
+          </Modal>
+        );
+      default:
+        return null;
+    }
+  }, [modalData]);
+
+  return renderModal();
+};
```

Comment:
consider splitting into separate modal components instead of huge switch/case

Reply:
it's already splited from page to separate component - MentorRegistryModal.tsx, it's used Modals from AntDesign and I'm really don't know how I can split it more than it is now  
Other point - it's not used in any other places in the app

Reply:
from what I see: 3 modals are inside the same component.

>Other point - it's not used in any other places in the app
it is not about reuse, while about readability and maintainability of 1 big component vs small dedicated components

Reply:
>switch case in pagecomponent to handle this modes

and this is right way to go. You don't need to have `callback` and call it the same function. The way it is not - not natural for react

Reply:
I suggest to extract each modal type to own component. The orchestration can be either mentorregistrymodal or inside a page, whatever makes sense

Reply:
even if each modal is just AntDesign component with different props?
It seems strange

Reply:
sure, this is how react components should be built - dump and simple. If it doesn't resonate with u, at least remove this `callback` which seems not needed

Reply:
+1, we could simply use components instead of raw tsx stuff in this switch at least  
And please consider to use `enum`  instead of plain values (I mean `invite`, `re-send` and `delete`)